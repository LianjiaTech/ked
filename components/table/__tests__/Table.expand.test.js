/* eslint-disable react/no-multi-comp */
import React from 'react';
import { mount } from 'enzyme';
import Table from '..';

const columns = [
  {
    title: 'Name',
    key: 'name',
    dataIndex: 'name',
  },
];

const data = [
  {
    key: '1',
    firstName: 'John',
    lastName: 'Brown',
    age: 32,

    children: [
      {
        key: '2',
        firstName: 'Jim',
        lastName: 'Green',
        age: 42,
      },
    ],
  },
];

describe('Table.expand', () => {
  it('click to expand', () => {
    const wrapper = mount(<Table columns={columns} dataSource={data} />);
    wrapper
      .find('.ant-table-row-expand-icon')
      .last()
      .simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should support expandIconColumnIndex', () => {
    const wrapper = mount(<Table columns={[]} dataSource={data} expandIconColumnIndex={1} />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
