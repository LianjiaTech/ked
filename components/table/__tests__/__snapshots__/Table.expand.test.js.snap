// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.expand click to expand 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-default ant-table-scroll-position-left"
      >
        <div
          class="ant-table-content"
        >
          <div
            class="ant-table-body"
          >
            <table
              class=""
            >
              <colgroup>
                <col />
              </colgroup>
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-row-cell-last"
                  >
                    <span
                      class="ant-table-header-column"
                    >
                      <div>
                        <span
                          class="ant-table-column-title"
                        >
                          Name
                        </span>
                        <span
                          class="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class=""
                  >
                    <span
                      class="ant-table-row-indent indent-level-0"
                      style="padding-left: 0px;"
                    />
                    <div
                      aria-label="Collapse row"
                      class="ant-table-row-expand-icon ant-table-row-expanded"
                      role="button"
                      tabindex="0"
                    />
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-1"
                  data-row-key="2"
                >
                  <td
                    class=""
                  >
                    <span
                      class="ant-table-row-indent indent-level-1"
                      style="padding-left: 20px;"
                    />
                    <span
                      class="ant-table-row-expand-icon ant-table-row-spaced"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination"
        unselectable="unselectable"
      >
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-prev"
          title="Previous Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              aria-label="icon: left"
              class="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-next"
          title="Next Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              aria-label="icon: right"
              class="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Table.expand should support expandIconColumnIndex 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-default ant-table-scroll-position-left"
      >
        <div
          class="ant-table-content"
        >
          <div
            class="ant-table-body"
          >
            <table
              class=""
            >
              <colgroup />
              <thead
                class="ant-table-thead"
              />
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                />
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination"
        unselectable="unselectable"
      >
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-prev"
          title="Previous Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              aria-label="icon: left"
              class="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-next"
          title="Next Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              aria-label="icon: right"
              class="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
